src/cobra/core/gene.py

```diff
@@ -299,3 +271,360 @@ def _repr_html_(self):
             n_reactions=len(self.reactions),
             reactions=format_long_string(", ".join(r.id for r in self.reactions), 200),
         )
+
+
+class GPR(Module):
+    """A Gene Reaction rule in a cobra model, using AST as base class.
+
+    Parameters
+    ----------
+    gpr_from : Expression or Module or AST
+        A GPR in AST format
+    """
+
+    def __init__(
+        self, gpr_from: Union[Expression, Module, AST] = None, *args, **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+        self._genes = set()
+        if gpr_from:
+            if isinstance(gpr_from, str):
+                self.from_string(gpr_from)
+                raise TypeError(
+                    f"GPR accepts AST, not string. "
+                    f'Next time, use GPR().from_string("{gpr_from}")'
+                )
+            elif isinstance(gpr_from, (Expression, Module)):
+                cleaner = GPRCleaner()
+                cleaner.visit(gpr_from)
+                self._genes = cleaner.gene_set
+                # noinspection PyTypeChecker
+                self.body = gpr_from.body
+                self.eval()
+            else:
+                raise (TypeError, "GPR requires AST Expression or Module")
+
+    @classmethod
+    def from_string(cls: type, string_gpr: str) -> "GPR":
```

Comment:
It's fine not to annotate `cls`.

```suggestion
    def from_string(cls, string_gpr: str) -> "GPR":
```